### ISSUE TITLE
[SEMANTIC HTML, SCREEN READER]: <dl> MUST have at least one <dt> element followed by at least one <dd> element

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#8515]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

> `<dl>` elements must only directly contain properly-ordered `<dt>` and `<dd>` groups, `<script>`, `<template>` or `<div>` elements

The use of the description list **must** have a term (`<dt>`). The use of `<span>` in the structure should be enhanced if replaced with a `<dfn>`.

In Internet Explorer and JAWS, the list context isn't conveyed, due to the lack of `dt`.

### Why it Matters

Screen readers have a specific way of announcing definition lists. When such lists are not properly marked up, this creates the opportunity for confusing or inaccurate screen reader output.

A definition list is used to provide the definitions of words or phrases. The Definition List is marked up using the dl element. Within the list, each term is put in a separate dt element, and its definition goes in the dd element directly following it.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to understand the content of the page with the same context provided to sighted users.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: any
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov/disability/view-dependents/ in browser
2. Log in as user 271
3. Open developer tools, and run axe-coconut scan
4. Verify the error documented in this ticket

### Current code
```html
<dl class="vads-l-row vads-u-background-color--gray-lightest vads-u-margin-top--0 vads-u-margin-bottom--2 vads-u-padding-top--1 vads-u-padding-bottom--2 vads-u-padding-x--2">
  <dd class="vads-l-col--12 vads-u-margin--0 vads-u-font-size--lg vads-u-font-weight--bold">SUE BROOKS</dd>
  <dd class="vads-l-col--12 vads-u-margin--0">
    <span class="vads-u-font-weight--bold">Relationship:</span> Spouse
  </dd>
  <dd class="vads-l-col--12 vads-u-margin--0">
    <span class="vads-u-font-weight--bold">Date of birth: </span>05/01/1955
  </dd>
</dl>
<!-- followed by dl's for each dependent -->
```

### Potential fix

```diff
<dl class="vads-l-row vads-u-background-color--gray-lightest vads-u-margin-top--0 vads-u-margin-bottom--2 vads-u-padding-top--1 vads-u-padding-bottom--2 vads-u-padding-x--2">

+  <dt class="vads-l-col--12 vads-u-margin--0 vads-u-font-size--lg vads-u-font-weight--bold">SUE BROOKS</dt>
  <dd class="vads-l-col--12 vads-u-margin--0">
+    <dfn class="vads-u-font-weight--bold">
      Relationship:
    </dfn> 
    Spouse
  </dd>
  <dd class="vads-l-col--12 vads-u-margin--0">
+    <dfn class="vads-u-font-weight--bold">
      Date of birth: 
    </dfn>
    05/01/1955
  </dd>

+  <dt class="vads-l-col--12 vads-u-margin--0 vads-u-font-size--lg vads-u-font-weight--bold">BILL BROOKS</dt>
  <dd class="vads-l-col--12 vads-u-margin--0">
+    <dfn class="vads-u-font-weight--bold">
      Relationship:
+    </dfn> 
    Parent
  </dd>
  <dd class="vads-l-col--12 vads-u-margin--0">
+    <dfn class="vads-u-font-weight--bold">
      SSN:
+    </dfn> 
    251456454
  </dd>
  <dd class="vads-l-col--12 vads-u-margin--0">
+    <dfn class="vads-u-font-weight--bold">
      Date of birth: 
+    </dfn>
    05/01/1919
  </dd>

<!-- and each additional dependent would have the name in a dt, and details in dd -->
</dl>
```

## WCAG or Vendor Guidance (optional)

* [W3C Using description lists](https://www.w3.org/WAI/WCAG21/Techniques/html/H40.html)

* [Deque University - `<dl>` elements must only directly contain properly-ordered `<dt>` and `<dd>` groups, `<script>`, or `<template>` elements](https://dequeuniversity.com/rules/axe/3.5/definition-list?application=AxeChrome)

## Screenshots or Trace Logs

![Screen Shot 2020-04-28 at 1 11 59 PM](https://user-images.githubusercontent.com/57469/80516718-e07f0b00-8951-11ea-99cc-5e5e2da53e7b.png)

**Comment by jenstrickland** [comment]
@micahchiang I'm sorry, I need to validate before we close these tickets.

**Comment by jenstrickland** [comment]
Hey @jason-gcio , do you know if Lighthouse API flipped a switch so the user data doesn't appear anymore? I'm using user 271trying to validate but the page says no dependents on file.

**Comment by jason-gcio** [comment]
Yes, they needed to switch it back...I will see if they can flip it and when.

**Comment by jason-gcio** [comment]
Lighthouse has indicated that this is still in motion but they do not have a date yet to say when this will work.  We are actively monitoring this but we don't yet have a specific ticket to show its completion.

**Comment by jason-gcio** [comment]
@jenstrickland - User 1 on this url should get us to where we need to be: https://staging.va.gov/view-change-dependents/view

**Comment by jenstrickland** [comment]
Validated that we have compliant HTML now. 

This page would also need to be remediated as noted in https://github.com/department-of-veterans-affairs/va.gov-team/issues/4637. Do you want me to write another ticket?

**Comment by jason-gcio** [comment]
Hmm, we don't have a slot open in the very near future to get to this but I have added it to the Hardening and Maintenance Epic for Rated Disabilities.  When we iterate on that this will be included then.  No need to open a new ticket, we can use #4367.
